lib/string_reverse.rb

```diff
@@ -1,6 +1,18 @@
 # A method to reverse a string in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n), where n is the number of elements in the array
+# Space complexity: O(1)
 def string_reverse(my_string)
-  raise NotImplementedError
+  return false if my_string == nil || my_string.length == 0
+
+  i = 0
+  j = my_string.length - 1
+  while i < j
+    temp = my_string[i]

```

Comment:
Tiny (personal preference) nit: some people (and companies) prefer more explanatory (but less concise) variable names than `i` and `j`.